modules/user/user-snapshot.service.test.ts

```diff
@@ -0,0 +1,1438 @@
+                            snapshots: [],
+                        }),
+                    );
+                }
+                // important, sort snapshots ASC
+                return res(
+                    ctx.data({
+                        snapshots: [
+                            {
+                                id: `${userAddress}-${threeDaysAgo}`,
+                                user: {
+                                    id: userAddress,
+                                },
+                                timestamp: threeDaysAgo,
+                                walletTokens: [poolAddress1],
+                                walletBalances: ['1'],
+                                gauges: [],
+                                gaugeBalances: [],
+                                farms: [],
+                                farmBalances: [],
+                            },
+                            {
+                                id: `${userAddress}-${twoDaysAgo}`,
+                                user: {
+                                    id: userAddress,
+                                },
+                                timestamp: twoDaysAgo,
+                                walletTokens: [],
+                                walletBalances: [],
+                                gauges: [],
+                                gaugeBalances: [],
+                                farms: [],
+                                farmBalances: [],
+                            },
+                        ],
+                    }),
+                );
+            }),
+        ],
+    );
+
+    const snapshotsFromService = await userService.getUserBalanceSnapshotsForPool(userAddress, poolId1, 'THIRTY_DAYS');
+    //check if 4th snapshot has been inferred from three present ones
+    expect(snapshotsFromService.length).toBe(2);
+    const snapshotsFromDb = await prisma.prismaUserPoolBalanceSnapshot.findMany({
+        where: {
+            userAddress: userAddress,
+        },
+        include: { pool: true },
+    });
+
+    // check if the 3 snapshots have been persisted
+    expect(snapshotsFromDb.length).toBe(2);
+
+    // check if balances are calculated correctly
+    expect(snapshotsFromService[0].timestamp).toBe(threeDaysAgo);
+    expect(snapshotsFromService[0].walletBalance).toBe('1');
+    expect(snapshotsFromService[1].timestamp).toBe(twoDaysAgo);
+    expect(snapshotsFromService[1].walletBalance).toBe('0');
+    expect(snapshotsFromService[1].totalValueUSD).toBe('0');
+    expect(snapshotsFromService[1].fees24h).toBe('0');
+    expect(snapshotsFromService[1].percentShare).toBe(0);
+});
+
+test('0 $ value user snapshots if there is no pool snapshot', async () => {
+    /*
+    Scenario: 
+    - The user requests the user stats for the first time
+    - The user joined pool2 three days ago and is still in the pool
+    - Poolsnapshots are only available for two days, therefore only two $ value > 0 snapshots are present
```

Comment:
so because we only store 2 days of pool snapshots, the first day he joined (3 days ago) has no $ values?